theseus/geometry/so3.py

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional, Union, cast
+
+import torch
+
+import theseus.constants
+
+from .lie_group import LieGroup
+from .point_types import Point3
+
+
+class SO3(LieGroup):
```

Comment:
The quaternion stuff will be useful in SE3 class and potentially in other places. Maybe pull it out as a separate class.